src/transformers/models/mistral/modeling_mistral.py

```diff
@@ -1075,6 +1041,100 @@ def forward(
             hidden_states=all_hidden_states,
             attentions=all_self_attns,
         )
+    
+    # copied from Llama implementation 
```

Comment:
We need to use 
```suggestion
    # Copied from transformers.models.llama.modeling_llama.LlamaForModel._update_causal_mask
```

Reply:
otherwise the copied from is not enforced

Reply:
actually I made some adaptions for sliding window inside, so this function here is not actually the same as that in llama